### PR TITLE
FIX #4573 : added custom fusion chart config to override default message

### LINKED ISSUE #4573: [Bug]When the User drag and drop the Chart widget->and select the custom chart from chart type->the chart type not supported text is displayed
## Description

When the User drag and drop the Chart widget->and select the custom chart from chart type->the "chart type not supported text" is displayed 

1.Go to Widgets Section
2.Drag and drop Chart Widget
3.On the floating window select the chart type as custom chart
4.verify the display of the Message displayed on the chart
Chartwidget

expected result: The Message should be changed to please enter chart data in chart data section as we have option to build the custom chart

actual result: The Message is displayed as "chart type not supported" which is not a valid message

### Important Details
 - Version: Cloud 
 - OS: Windows
 - Browser Chrome

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description
When the User drag and drop the Chart widget->and select the custom chart from chart type->the "chart type not supported text" is displayed. Added a default config to resolve it.

Fixes #4573 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

File: app/client/src/mockResponses/WidgetConfigResponse.tsx [hunk 1]
```diff
@@ -541,6 +541,12 @@ const WidgetConfigResponse: WidgetConfigReducerState = {
       },
       xAxisName: "Last Week",
       yAxisName: "Total Order Revenue $",
+      customFusionChartConfig: {
+        config: {
```
Comment (somangshu) [inline app/client/src/mockResponses/WidgetConfigResponse.tsx:545]: Today, the custom chart data does not have a config property in the object.

File: app/client/src/mockResponses/WidgetConfigResponse.tsx [hunk 2]
```diff
@@ -541,6 +541,12 @@ const WidgetConfigResponse: WidgetConfigReducerState = {
...
+          type: "",
```
Comment (somangshu) [inline app/client/src/mockResponses/WidgetConfigResponse.tsx:546]: Even if we add the empty object like this, the message `chart data is not supported` will still exist, We could add a sample config here like other chart types to show an example of usage. Check out the [docs](https://docs.appsmith.com/widget-reference/chart)

@riodeuno thoughts?

**Review by somangshu — CHANGES_REQUESTED** [review]

**Comment by somangshu** [comment]
Does not work on the old app charts!

@riodeuno does this change work according to you? We are now showing an example config preloaded in the custom chart data field.

File: app/client/src/mockResponses/WidgetConfigResponse.tsx (hunk 2 above)
Comment (riodeuno) [reply app/client/src/mockResponses/WidgetConfigResponse.tsx]: Yes, at the very least we should include the `type` property to make sure a chart is visible. We can also copy the data from the `chartData` property's default and use it here in the `dataSource` property.

File: app/client/src/mockResponses/WidgetConfigResponse.tsx [hunk 3]
```diff
@@ -541,6 +541,52 @@ const WidgetConfigResponse: WidgetConfigReducerState = {
...
+            caption: "Monthly revenue for last year",
+            subCaption: "Harry's SuperMart",
+            xAxisName: "Month",
+            yAxisName: "Revenues (In USD)",
+            numberPrefix: "$",
+            theme: "fusion",
+          },
+          data: [
+            {
+              label: "Jan",
```
Comment (riodeuno) [inline app/client/src/mockResponses/WidgetConfigResponse.tsx:557]: Shall we make this data and the data above in the `chartData` property consistent?

Comment (somangshu) [reply app/client/src/mockResponses/WidgetConfigResponse.tsx:557]: @vibhandikyash can you look into this

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
Rerunning jobs, Failing for some reason

**Pull request merged**